cumulusci/core/flowrunner.py

```diff
@@ -628,33 +630,37 @@ def _check_old_yaml_format(self):
             else:
                 raise FlowConfigError("No steps found in the flow definition")
 
-    def _check_infinite_flows(self, flow_config, visited_flows=None):
+    def _check_infinite_flows(self, flow_config, flow_stack=None):
         """
         Recursively loop through the flow_config and check if there are any cycles.
 
-        :param steps: Set of step definitions to loop through
-        :param flows: Flows already visited.
+        :param flow_config: Set of step definitions to loop through
+        :param flow_stack: list of flow signatures already visited
         :return: None
         """
-        if visited_flows is None:
-            visited_flows = set()
+        if flow_stack is None:
```

Comment:
`None` is also a "falsy" value so you can shorten this to:
```suggestion
        if not flow_stack:
```